packages/web-frontend/cypress/scripts/generateConfig/VisualisationFilter.js

```diff
@@ -70,11 +83,31 @@ class VisualisationFilter {
 }
 
 export class DashboardReportFilter extends VisualisationFilter {
-  fetchRecords = async objectUrls =>
-    this.db.find('dashboardReport', {
-      id: objectUrls.map(u => u.id),
-      drillDownLevel: null, // Drill down reports are not currently supported in e2e tests
+  fetchRecords = async objectUrls => {
+    const dashboardItems = await this.db.find('dashboard_item', {
+      code: objectUrls.map(u => u.code),
+    });
+    const legacyDashboardItems = dashboardItems.filter(di => di.legacy);
+    const legacyReports = await this.db.find('legacy_report', {
+      code: legacyDashboardItems.map(di => di.report_code),
+    });
+    const legacyReportById = keyBy(legacyReports, 'code');
+    return dashboardItems.map(di => {
+      const legacyReport = di.legacy ? legacyReportById[di.report_code] : {};
+
+      if (legacyReport) {
+        return {
+          ...di,
+          data_builder: legacyReport.data_builder,
+          data_builder_config: legacyReport.data_builder_config,
+        };
+      }
```

Comment:
Information from the old `dashboardReport` table is now splitted into `dashboard_item` and `legacy_report`. So I feel it's easier for the filter `fetchRecords()` to just return a wrapper object of `dashboard_item` and `legacy_report`. We can do the same for the modern `report` table, but I feel that there's nothing really helpful to filter from `report` table at the moment...

Reply:
Good call Chris! Moreover - with the new way of running the tests (against a deployed instance, not localhost) unfortunately most of the time we won't be able to (accurately) use the filters, since they will be using the DB on the host machine, not on the testable instance. For CI/CD we don't even have a DB with actual data, so using filters there would be impossible at the moment.

Which means that all the fancy filtering functionality that was added in the last PR will most probably be unused, at least for a while 😆 

Still, thanks for taking the time to update this!